backend/src/services/comparison_service.py

```
@@ -1030,57 +1475,128 @@ def _identify_metric_outliers(
 
-            # Generate mock correlation
-            coefficient = np.random.uniform(-0.8, 0.8)
-            p_value = np.random.uniform(0, 0.1)
+                    # Match entities by ID and create paired values
+                    other_values_dict = {e.id: e.value for e in other_entities}
+
+                    # Build paired arrays
+                    paired_primary = []
+                    paired_other = []
+
+                    for entity_id, primary_val in zip(entity_ids, primary_values):
```

Ruff (0.14.3)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)